src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java

```diff
@@ -22,6 +23,20 @@ public UserDaoImpl(SessionFactory sessionFactory) {
                     + "WHERE u.email = :email", User.class);
             query.setParameter("email", email);
             return query.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get user by email: " + email, e);
+        }
+    }
+
+    @Override
+    public Optional<User> getById(Long userId) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<User> getUserQuery = session.createQuery(
+                    "FROM User u WHERE u.id = :id", User.class);
+            getUserQuery.setParameter("id", userId);
+            return Optional.ofNullable(getUserQuery.getSingleResult());
```

Comment:
return Optional.ofNullable(session.get(User.class, userId));

Reply:
Why it mistake? I think my solution is good

Reply:
Both options are suitable, but using session.get the code will be shorter